pandas/core/internals.py

```diff
@@ -1026,7 +1026,7 @@ def f(m, v, i):
 
         return [self.make_block(new_values, fastpath=True)]
 
-    def coerce_to_target_dtype(self, other):
+    def coerce_to_target_dtype(self, other, force_coericion=False):
```

Comment:
huh? I don't think you should be touching this at all, this is much more related to some logic in ``ops.py``

Reply:
Looks like the problem is that before the operation is done, `coerce_to_target_dtype` coerces the `np.array` to `object` in this block:

    if self.is_bool or is_object_dtype(dtype) or is_bool_dtype(dtype):
        # we don't upcast to bool
         return self.astype(object)

Example here:

    >>> import numpy as np
    >>> (np.array([True]) * 1.0).dtype
    dtype('float64')
    >>> (np.array([True], dtype='object') * 1.0).dtype
    dtype('O')

Reply:
you are missing the point; the fix is much higher up the stack